File: frontend/classes/Title.ts
interface TitleData {
  id: number
  senator: number
  name: string
  start_step: number
  end_step: number | null
  major_office: boolean
}

class Title {
  id: number
  senator: number
  name: string
  start_step: number
  end_step: number | null
  major_office: boolean

  constructor(data: TitleData) {
    this.id = data.id
    this.name = data.name
    this.senator = data.senator
    this.start_step = data.start_step
    this.end_step = data.end_step
    this.major_office = data.major_office
  }
}

export default Title

File: backend/rorapp/models/title.py
from django.db import models
from rorapp.models.senator import Senator
from rorapp.models.step import Step


# Model for representing titles
class Title(models.Model):
    name = models.CharField(max_length=21)
    senator = models.ForeignKey(Senator, on_delete=models.CASCADE)
    start_step = models.ForeignKey(Step, on_delete=models.CASCADE, related_name='starting_title_set')
    end_step = models.ForeignKey(Step, on_delete=models.CASCADE, blank=True, null=True, related_name='ending_title_set')  # Null means the title is active
    major_office = models.BooleanField(default=False)
    
    # String representation of the title, used in admin site
    def __str__(self):
        return f'{self.id}: {self.name} {self.senator}'
